### PR TITLE
CI: Test migrations

### PR DESCRIPTION
**Pull request by 4ydan**
#134

## Basics

- [x] I added a line to [/doc/CHANGELOG.md](/doc/CHANGELOG.md)
- [x] The PR is rebased with current master.
- [x] Details of what you changed are in commit messages.
- [x] References to issues, e.g. `close #X`, are in the commit messages.
- [x] The buildserver is happy.

## Checklist

- [ ] I have installed and I am using [pre-commit hooks](../doc/contrib/README.md#Hooks)
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I fixed all affected decisions
- [ ] I added unit tests for my code
- [ ] I added code comments, logging, and assertions as appropriate
- [ ] I translated all strings visible to the user
- [ ] I mentioned [every code or binary](/.reuse/dep5) not directly written or done by me in [reuse syntax](https://reuse.software/)
- [ ] I created left-over issues for things that are still to be done
- [ ] Code is conforming to [our Architecture](/doc/architecture)
- [ ] Code is conforming to [our Guidelines](/doc/guidelines)
      (exceptions are documented)
- [ ] Code is consistent to [our Design Decisions](/doc/decisions)

## Review

- [ ] I've tested the code
- [ ] I've read through the whole code
- [ ] Examples are well chosen and understandable

### CODE REVIEW

File: ci/Jenkinsfile [hunk 1]
```diff
@@ -85,6 +85,10 @@ def testAndBuildBackend() {
 
                     stage('build-backend'){
                         sh './ci/build-scripts/build-backend.sh'
+                        sh 'cd backend && diesel migration redo'
+                        if (env.BRANCH_NAME == 'master') {
+                            sh 'cd backend && diesel migration redo -a'
```
Comment (markus2330) [inline ci/Jenkinsfile:90]: Isn't this logic inverted? The `redo -a` should be only done for PRs.

Any reason why you don't use the Makefile targets here?

**Review by markus2330 — CHANGES_REQUESTED** [review]
Good to finally get this! :rocket:

File: ci/Jenkinsfile (hunk 1 above)
Comment (4ydan) [reply ci/Jenkinsfile:90]: I was thinking to refactor all to make in one PR, but can already start I guess.

**Comment by 4ydan** [comment]
Can we do the migration check before building?

**Comment by 4ydan** [comment]
image
Ok so now we need proper rights, is that to be fixed in a separate PR and we wait for the merge?

**Comment by markus2330** [comment]
> Ok so now we need proper rights, is that to be fixed in a separate PR and we wait for the merge?

I don't think it is a permission problem. It rather looks like the migrations are broken: the PostGIS extension is tried to be removed but not yet all objects that use PostGIS are removed. (At least I interpret the error message like this.)

I tried to reproduce and got another problem. I created #674 to track these issues.

Probably, we need to wait for #644 (or even longer) until these problems are resolved...

**Comment by markus2330** [comment]
jenkins build please

**Comment by markus2330** [comment]
Let us see if #644 improved the situation...

**Comment by markus2330** [comment]
jenkins build please

**Comment by markus2330** [comment]
We again have `Failed to run 2023-07-01-152628_squashed_setup with: cannot drop extension postgis because other objects depend on it` here. I cannot reproduce it locally. Maybe it is a permission problem after all?

@temmey do you maybe know what this error is?

**Comment by 4ydan** [comment]
Changing `DROP EXTENSION postgis`; to `DROP EXTENSION postgis CASCADE`; solves the problem, but not sure if we want this. I also can't reproduce it locally.

**Comment by markus2330** [comment]
Great that you found a way!

According to https://www.postgresql.org/docs/current/sql-dropextension.html

> Automatically drop objects that depend on the extension, and in turn all objects that depend on those objects (see [Section 5.14](https://www.postgresql.org/docs/current/ddl-depend.html)).

As we also reset the whole database on **pr** builds, afaik there cannot be any harm be removing any objects in the database. On **dev** (master) or production builds, it might be a problem but there we don't test `redo -a` anyway and then it doesn't occur, does it?

**Comment by 4ydan** [comment]
Yes I agree, but somehow it still bothers me that there is something.

>  it might be a problem but there we don't test redo -a anyway and then it doesn't occur, does it?

True, it only occured with `make migration-redo-a`.

**Comment by 4ydan** [comment]
I guess it will help more than it could do bad. Because finding the issue for that error might take some time in which other PR's could fail migrations maybe?

File: doc/CHANGELOG.md [hunk 2]
```diff
@@ -50,6 +50,7 @@ Syntax: `- short text describing the change _(Your Name)_`
 - _()_
 - CI: Improve pipeleine parallelization #577 _(4ydan)_
 - CI: Optimized mdbook docker image #585 _(4ydan)_
+- CI: Test migrations in PR and master #134 _(4ydan)_
```
Comment (markus2330) [inline doc/CHANGELOG.md:53]:
```suggestion
- CI: Test `diesel` migrations in PR and master #134 _(4ydan)_
```

**Comment by markus2330** [comment]
Yes, let us merge this, a rebase is needed, though.

I don't think an issue is needed. It is okay if data gets lost on pr or dev.

**Comment by 4ydan** [comment]
Ready to be merged.

**Comment by markus2330** [comment]
Thank you! :heart:

**Pull request merged**